envisage/tests/ets_config_patcher.py

```diff
@@ -0,0 +1,61 @@
+# (C) Copyright 2007-2019 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+
+import os
+import shutil
+import tempfile
+
+
+class ETSConfigPatcher(object):
+    """
+    Object that patches the directories in ETSConfig, to avoid having
+    tests write to the home directory.
+    """
+    def __init__(self):
+        from traits.etsconfig.api import ETSConfig
+        self.etsconfig = ETSConfig
+
+        self.tmpdir = None
+        self.old_application_data = None
+        self.old_application_home = None
+        self.old_user_data = None
+
+    def start(self):
+        tmpdir = self.tmpdir = tempfile.mkdtemp()
+        self.etsconfig._application_data = os.path.join(
+            self.tmpdir,
+        )
```

Comment:
i'm not a 100% sure what this code is doing. copy-paste error?

Reply:
Yep; looks like I messed up when moving lines around. Fixed.